File: my_app/lib/model/Alimento.dart
class Alimento {
  final String name;
  final double cantidad;
  final String unidadesCantidad;
  final double calorias;
  final double grasas;
  final double proteinas;
  final double carbohidratos;
  final double sodio;
  final double azucar;
  final double fibra;
  final String image;

  Alimento({
    required this.name,
    required this.cantidad,
    required this.unidadesCantidad,
    required this.calorias,
    required this.grasas,
    required this.proteinas,
    required this.carbohidratos,
    required this.azucar,
    required this.fibra,
    required this.sodio,
    required this.image,
  });

  factory Alimento.fromJson(Map<String, dynamic> json) {
    return Alimento(
      name: json['name'] as String,
      cantidad: json['cantidad'] as double,
      unidadesCantidad: json['unidadesCantidad'] as String,
      calorias: json['calorias'] as double,
      grasas: json['grasas'] as double,
      proteinas: json['proteinas'] as double,
      carbohidratos: json['carbohidratos'] as double,
      azucar: json['azucar'] as double,
      fibra: json['fibra'] as double,
      sodio: json['sodio'] as double,
      image: json['image'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'cantidad': cantidad,
      'unidadesCantidad': unidadesCantidad,
      'calorias': calorias,
      'grasas': grasas,
      'proteinas': proteinas,
      'carbohidratos': carbohidratos,
      'sodio': sodio,
      'azucar': azucar,
      'fibre': fibra,
      'image': image,
    };
  }
}

File: nutri/src/main/java/com/demo/nutri/model/Alimento.java
package com.demo.nutri.model;


import java.net.URI;

import javax.persistence.*;

@Entity
public class Alimento {
    @Id
    @GeneratedValue
    private Integer id;
    private String name;
    private Double cantidad;
    private String unidadesCantidad; 
    private Double calorias;
    private Double grasas;
    private Double proteinas;
    private Double carbohidratos;
    private Double azucar;
    private Double fibra;
    private Double sodio;
    private String image; 
    private String nombreUsuario; 
    private String codigoDeBarras;
    @ManyToOne
    private Receta receta;
    @OneToOne
    private RegistroComidas registroComidas;

    public String getCodigoDeBarras() {
        return codigoDeBarras;
    }
    public void setCodigoDeBarras(String codigoDeBarras) {
        this.codigoDeBarras = codigoDeBarras;
    }
    
    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public Double getCantidad() {
        return cantidad;
    }
    public void setCantidad(Double cantidad) {
        this.cantidad = cantidad;
    }
    public String getUnidadesCantidad() {
        return unidadesCantidad;
    }
    public void setUnidadesCantidad(String unidadesCantidad) {
        this.unidadesCantidad = unidadesCantidad;
    }
    public Double getCalorias() {
        return calorias;
    }
    public void setCalorias(Double calorias) {
        this.calorias = calorias;
    }
    public Double getGrasas() {
        return grasas;
    }
    public void setGrasas(Double grasas) {
        this.grasas = grasas;
    }
    public Double getProteinas() {
        return proteinas;
    }
    public void setProteinas(Double proteinas) {
        this.proteinas = proteinas;
    }
    public Double getCarbohidratos() {
        return carbohidratos;
    }
    public void setCarbohidratos(Double carbohidratos) {
        this.carbohidratos = carbohidratos;
    }
    public Double getAzucar() {
        return azucar;
    }
    public void setAzucar(Double azucar) {
        this.azucar = azucar;
    }
    public Double getFibra() {
        return fibra;
    }
    public void setFibra(Double fibra) {
        this.fibra = fibra;
    }
    public Double getSodio() {
        return sodio;
    }
    public void setSodio(Double sodio) {
        this.sodio = sodio;
    }
    public String getImage() {
        return image;
    }
    public void setImage(String image) {
        this.image = image;
    }
    public String getNombreUsuario() {
        return nombreUsuario;
    }
    public void setNombreUsuario(String nombreUsuario) {
        this.nombreUsuario = nombreUsuario;
    }
    public Receta getReceta() {
        return receta;
    }
    public void setReceta(Receta receta) {
        this.receta = receta;
    }
    public RegistroComidas getRegistroComidas() {
        return registroComidas;
    }
    public void setRegistroComidas(RegistroComidas registroComidas) {
        this.registroComidas = registroComidas;
    }




}




 
